crates/criticaltrust/src/signatures/payload.rs

```diff
@@ -160,138 +162,139 @@ mod tests {
 
-        let expired = test_env.create_key_with_expiry(KeyRole::Packages, -1);
-        assert_verify_fail(&test_env, &[&expired]);
+        let expired = test_env.create_key_with_expiry(KeyRole::Packages, -1).await;
+        assert_verify_fail(&test_env, &[&expired]).await;
     }
 
-    #[test]
-    fn test_verify_with_not_expired_key() {
-        let mut env = TestEnvironment::prepare();
+    #[tokio::test]
+    async fn test_verify_with_not_expired_key() {
+        let mut env = TestEnvironment::prepare().await;
 
-        let not_expired = env.create_key_with_expiry(KeyRole::Packages, 1);
-        assert_verify_pass(&env, &[&not_expired]);
+        let not_expired = env.create_key_with_expiry(KeyRole::Packages, 1).await;
+        assert_verify_pass(&env, &[&not_expired]).await;
     }
 
-    #[test]
-    fn test_verify_with_expired_and_not_expired_keys() {
-        let mut test_env = TestEnvironment::prepare();
+    #[tokio::test]
+    async fn test_verify_with_expired_and_not_expired_keys() {
+        let mut test_env = TestEnvironment::prepare().await;
 
-        let expired = test_env.create_key_with_expiry(KeyRole::Packages, -1);
-        let not_expired = test_env.create_key_with_expiry(KeyRole::Packages, 1);
+        let expired = test_env.create_key_with_expiry(KeyRole::Packages, -1).await;
+        let not_expired = test_env.create_key_with_expiry(KeyRole::Packages, 1).await;
 
-        assert_verify_pass(&test_env, &[&expired, &not_expired]);
-        assert_verify_pass(&test_env, &[&not_expired, &expired]);
+        assert_verify_pass(&test_env, &[&expired, &not_expired]).await;
+        assert_verify_pass(&test_env, &[&not_expired, &expired]).await;
     }
 
     // Signature
 
-    #[test]
-    fn test_verify_with_bad_signature() {
-        let mut test_env = TestEnvironment::prepare();
+    #[tokio::test]
+    async fn test_verify_with_bad_signature() {
+        let mut test_env = TestEnvironment::prepare().await;
 
-        let bad = BadKeyPair(test_env.create_key(KeyRole::Packages));
-        assert_verify_fail(&test_env, &[&bad]);
+        let bad = BadKeyPair(test_env.create_key(KeyRole::Packages).await);
+        assert_verify_fail(&test_env, &[&bad]).await;
     }
 
-    #[test]
-    fn test_verify_with_bad_and_good_signature() {
-        let mut test_env = TestEnvironment::prepare();
+    #[tokio::test]
+    async fn test_verify_with_bad_and_good_signature() {
+        let mut test_env = TestEnvironment::prepare().await;
 
-        let bad = BadKeyPair(test_env.create_key(KeyRole::Packages));
-        let good = test_env.create_key(KeyRole::Packages);
-        assert_verify_pass(&test_env, &[&bad, &good]);
-        assert_verify_pass(&test_env, &[&good, &bad]);
+        let bad = BadKeyPair(test_env.create_key(KeyRole::Packages).await);
+        let good = test_env.create_key(KeyRole::Packages).await;
+        assert_verify_pass(&test_env, &[&bad.0, &good]).await;
```

Comment:
We can observe a consequence of some of these API changes here. We can no longer pass differing `dyn &KeyPair` here. They need to be the same type.